fractal_server/app/routes/api/v2/task_collection.py

```diff
@@ -119,8 +121,39 @@ async def collect_tasks_pip(
     except FileExistsError:
         venv_path = create_package_dir_pip(task_pkg=task_pkg, create=False)
         try:
-            task_collect_status = get_collection_data(venv_path)
-            for task in task_collect_status.task_list:
+            package_path = get_absolute_venv_path(venv_path)
+            collection_path = get_collection_path(package_path)
+            with collection_path.open() as f:
+                task_collect_data = json.load(f)
+
+            if not isinstance(task_collect_data, dict):
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=(
+                        f"Collection '{str(collection_path)}' is not a Python "
```

Comment:
Cannot collect package, possible reason: an old version of the same package has already been collected.{str(collection_path)} has invalid content (not a Python dictionary)"